tests/ci/check_so_interface.py

```diff
@@ -0,0 +1,352 @@
+                ]
+            }
+        }
+    },
+    "python/pysmurf/client/debug/smurf_noise.py": {
+        "SmurfNoiseMixin": {
+            "methods" : [
+                'take_noise_psd', 'turn_off_noisy_channels', 'noise_vs_tone', 'noise_vs_bias', 'noise_vs_amplitude', 'noise_vs', 'get_datafiles_from_file', 'get_biases_from_file', 'get_iv_data', 'get_si_data', 'get_NEI_to_NEP_factor', 'analyze_noise_vs_bias', 'analyze_psd', 'noise_all_vs_noise_solo', 'analyze_noise_all_vs_noise_solo', 'NET_CMB', 'analyze_noise_vs_tone', 'take_noise_high_bandwidth', 'noise_svd', 'plot_svd_summary', 'plot_svd_modes', 'remove_svd'
+            ],
+            "take_noise_psd": {
+                "args": [
+                    'self', 'meas_time', 'channel', 'nperseg', 'detrend', 'fs', 'low_freq', 'high_freq', 'make_channel_plot', 'make_summary_plot', 'save_data', 'show_plot', 'grid_on', 'datafile', 'downsample_factor', 'write_log', 'reset_filter', 'reset_unwrapper', 'return_noise_params', 'plotname_append'
+                ],
+                "defaults": [
+                    'None', '2 ** 12', "'constant'", 'None', 'None', 'None', 'True', 'True', 'False', 'False', 'False', 'None', 'None', 'True', 'True', 'True', 'False', "''"
+                ]
+            }
+        },
+    }
+}
+
+"""
+Frozen Functions:
+
+SETUP FUNCTIONS
+setup
+set_amplifier_bias #smurf_command
+set_cryo_card_ps_en #smurf_command
+which_on #smurf_util
+band_off #smurf_util
+channel_off #smurf_util
+
+TUNING FUNCTIONS
+full_band_resp #smurf_tune.py
+find_freq #smurf_tune.py
+setup_notches #smurf_tune.py
+run_serial_gradient_descent #smurf_command
+run_serial_eta_scan #smurf_command
+plot_tune_summary #smurf_tune.py
+tracking_setup #smurf_tune.py
+set_amplitude_scale_array #smurf_command
+
+TES/FLUX RAMP FUNCTIONS
+set_tes_bias_bipolar_array #smurf_util
+set_tes_bias_high_current #smurf_util
+set_tes_bias_low_current #smurf_util
+set_mode_dc #smurf_util
+set_mode_ac #smurf_util
+flux_ramp_setup #smurf_tune.py
+
+DATA ACQUISITION FUNCTIONS
+set_stream_enable #smurf_command
+take_stream_data #smurf_util
+take_noise_psd #smurf_noise
+stream_data_on #smurf_util
+stream_data_off #surf_util
+read_stream_data #smurf_util
+set_downsample_filter #smurf_util
+
+IV FUNCTIONS
+run_iv #smurf_iv
+analyze_iv #smurf_iv
+
+DATA OUTPUTS TO DISK
+tune files generated when new resonators are found
+channel mapping file format
+.dat noise files - generated by take_stream_data
+iv_files - generated by run_iv"
+"""
+def compare_args(node, fname, classname, funcname, type):
+    # Check function
+    if type == "isfunc":
+        spec_args = interface[fname][classname][funcname]["args"]
+        spec_defaults = interface[fname][classname][funcname]["defaults"]
+
+        # Extract function arguments from the AST node
+        found_args = [arg.arg for arg in node.args.args]
```

Comment:
This looks good, but you may want to think about some more complicated cases that might occur. See [here](https://docs.python.org/3/library/ast.html#ast.arg) for an example of all the different types of arguments that can occur. We probably don't need to worry about positional-only or kw-only args because that is a pretty unusual pattern, but at least checking for `vararg` and `kwarg` would be good.